src/algorithms/reco/JetReconstruction.cc

```diff
@@ -90,36 +95,21 @@ namespace eicrecon {
 
       m_log->trace("  jet {}: pt = {}, y = {}, phi = {}", i, jets[i].pt(), jets[i].rap(), jets[i].phi());
 
-      // Type = 0 for jets, Type = 1 for constituents
-      // Use PDG values to match jets and constituents
+      // create jet to store in output collection
       edm4eic::MutableReconstructedParticle jet_output = jet_collection->create();
-      jet_output.setType(0);
-      jet_output.setPDG(i);
       jet_output.setMomentum(edm4hep::Vector3f(jets[i].px(), jets[i].py(), jets[i].pz()));
       jet_output.setEnergy(jets[i].e());
       jet_output.setMass(jets[i].m());
 
-      // loop over constituents
+      // link constituents to jet kinematic info
       std::vector<PseudoJet> csts = jets[i].constituents();
       for (unsigned j = 0; j < csts.size(); j++) {
-
-        const double cst_pt = csts[j].pt();
-        m_log->trace("    constituent {}'s pt: {}", j, cst_pt);
-
-        // Type = 0 for jets, Type = 1 for constituents
-        // Use PDG values to match jets and constituents
-        edm4eic::MutableReconstructedParticle cst_output = jet_collection->create();
-        cst_output.setType(1);
-        cst_output.setPDG(i);
-        cst_output.setMomentum(edm4hep::Vector3f(csts[j].px(), csts[j].py(), csts[j].pz()));
-        cst_output.setEnergy(csts[j].e());
-        cst_output.setMass(csts[j].m());
-        //jet_output.addToParticles(cst_output);  // FIXME: global issue with podio reference
+        add_to_jet_kinematics(input_collection[csts[j].user_index()], jet_output);
       } // for constituent j
     } // for jet i
 
     // return the jets
     return jet_collection;
-  }
+  }  // end 'process(const T&)'
```

Comment:
The style creep of jet reco code is not left unnoticed ;)

Reply:
😉